tests/rptest/services/redpanda.py

```diff
@@ -1560,8 +1560,38 @@ def get_node_by_id(self, id):
                 return p
         return None
 
-    def stop_node(self, node, **kwargs):
-        pass
+    def stop_node(self,
```

Comment:
For `RedpandaService` `stop_node` will stpo the node for good, right?

Here `stop_node` is more like "restart pod" since k8s will restart the pod. So I think we should call it that and not try to override the `stop_node` call which doesn't make sense for `RedpandaServiceCloud`. 

This would dovetail with my change in progress where `RedpandaServiceCloud` will no longer derive from `Service` so we aren't tied to that abstraction.

Reply:
i renamed `stop_node()` to `restart_pod()` in 4b70593c00d927dff4afdbe6eebc164db8321e9b and updated usage in 4bfa0c2ca57b286ca2b96697ee19779f618107a2 and d79e57eed21de67af512f8c248e017b0a8af0914